contribs/drt/src/main/java/org/matsim/contrib/drt/util/stats/DrtVehicleOccupancyProfileCalculator.java

```diff
@@ -98,16 +104,19 @@ public DrtVehicleOccupancyProfileCalculator(String dvrpMode, FleetSpecification
 				.orElse(0);
 	}
 
-	public void consolidate() {
-		for (VehicleState state : vehicleStates.values()) {
-			if (state.taskType != null) {
-				increment(state, analysisEndTime);
+	private void consolidate() {
+		if (!wasConsolidatedInThisIteration) {
+			for (VehicleState state : vehicleStates.values()) {
+				if (state.taskType != null) {
+					increment(state, analysisEndTime);
+				}
 			}
-		}
-		vehicleStates.clear();
+			vehicleStates.clear();
 
-		nonPassengerServingTaskProfiles.values().forEach(this::normalizeProfile);
-		vehicleOccupancyProfiles.forEach(this::normalizeProfile);
+			nonPassengerServingTaskProfiles.values().forEach(this::normalizeProfile);
+			vehicleOccupancyProfiles.forEach(this::normalizeProfile);
+			wasConsolidatedInThisIteration = true;
+		}
```

Comment:
Nice that now the client code is not need to worry about calling the consolidation. However, it still needs to be aware that the consolidation is run once per iteration (after a qsim ends), so calling get...() in the middle of a qsim will lead to errors.

Maybe adding some comments would be helpful.

Another option: require `!wasConsolidatedInThisIteration` inside all `handleEvent()` methods.

Reply:
I solved this now using an `AfterMobsimListener`, so consolidate() checks whether the `AfterMobsimEvent` has already been thrown and fails otherwise. Those `get...()` methods all call consolidate() in the end, so checking at consolidate() should be sufficient. This seems more computation time efficient than checking at each `handleEvent()` method. There should not be any ordering issues since the output analysis is called with the `IterationEndsEvent` which occurs after the `AfterMobsimEvent`.

Reply:
Looks really nice! Thanks!